src/TuDa.CIMS.Web/Components/SubmitPopup.razor.cs

```diff
@@ -0,0 +1,305 @@
+using Microsoft.AspNetCore.Components;
+using MudBlazor;
+using TuDa.CIMS.Shared.Entities;
+using TuDa.CIMS.Shared.Entities.Enums;
+
+namespace TuDa.CIMS.Web.Components;
+
+public partial class SubmitPopup
+{
+    [Inject] private IDialogService DialogService { get; set; }
+
+    private static Professor prof1 = new()
+    {
+        Name = "Prof. Mustermann",
+        WorkingGroup = new WorkingGroup()
+        {
+            Id = Guid.Empty,
+            PhoneNumber = "",
+            Professor = prof1,
+            Purchases = [],
+            Students = []
+        }
+    };
+
+    private static Professor prof2 = new()
+    {
+        Name = "Prof. Schmidt",
+        WorkingGroup = new WorkingGroup()
+        {
+            Id = Guid.Empty,
+            PhoneNumber = "",
+            Professor = prof1,
+            Purchases = [],
+            Students = []
+        }
+    };
+
+    private static Professor prof3 = new()
+    {
+        Name = "Prof. Müller",
+        WorkingGroup = new WorkingGroup()
+        {
+            Id = Guid.Empty,
+            PhoneNumber = "",
+            Professor = prof1,
+            Purchases = [],
+            Students = []
+        }
+    };
+
+    private static Professor prof4 = new()
+    {
+        Name = "Prof. Hofmann",
+        WorkingGroup = new WorkingGroup()
+        {
+            Id = Guid.Empty,
+            PhoneNumber = "",
+            Professor = prof1,
+            Purchases = [],
+            Students = []
+        }
+    };
+
+    private static Professor prof5 = new()
+    {
+        Name = "Prof. Meyer",
+        WorkingGroup = new WorkingGroup()
+        {
+            Id = Guid.Empty,
+            PhoneNumber = "",
+            Professor = prof1,
+            Purchases = [],
+            Students = []
+        }
+    };
+
+    /// <summary>
+    /// List of the professors.
+    /// </summary>
+
+    [Parameter]
+    public List<Professor> Professors { get; set; } =
+    [
+        prof1,
+        prof2, prof3, prof4, prof5
+    ];
+
+    /// <summary>
+    /// List of the names of the professors.
+    /// </summary>
+    private List<string> ProfessorNames { get; set; } =
+    [
+        prof1.Name,
+        prof2.Name, prof3.Name, prof4.Name, prof5.Name
+    ];
+
+    /// <summary>
+    /// Search for the selection of the working group.
+    /// </summary>
+
+    private Task<IEnumerable<string>> Search(string searchText, CancellationToken cancellationToken)
+    {
+        return Task.FromResult(ProfessorNames
+            .Where(prof =>
+                string.IsNullOrWhiteSpace(searchText) ||
+                prof.Contains(searchText, StringComparison.OrdinalIgnoreCase)));
+    }
+    /// <summary>
+    /// CascadingParamter MudDialag.
+    /// </summary>
+    [CascadingParameter] private MudDialogInstance MudDialog { get; set; }
+
+    /// <summary>
+    /// Closes the MudDialog.
+    /// </summary>
+
+    private void Submit() => MudDialog.Close(DialogResult.Ok(true));
```

Comment:
🔧 We should return the result of the dialog. So the selected working group. #95 provides an idea.